components/server/src/bitbucket-server/bitbucket-server-api.ts

```diff
@@ -283,29 +283,29 @@ export class BitbucketServerApi {
      * https://developer.atlassian.com/server/bitbucket/rest/v811/api-group-repository/#api-api-latest-repos-get
      */
     async getRepos(
-        user: User,
+        userOrToken: User | string,
         query: {
             permission?: "REPO_READ" | "REPO_WRITE" | "REPO_ADMIN";
-            /**
-             * defaults to 25
-             */
-            limit: number;
         },
     ) {
-        let q = "";
-        if (query) {
-            const segments = [];
-            if (query.permission) {
-                segments.push(`permission=${query.permission}`);
-            }
-            if (query.limit) {
-                segments.push(`limit=${query.limit}`);
+        const result: BitbucketServer.Repository[] = [];
+        const permission = query.permission ? `permission=${query.permission}&` : "";
+        let isLastPage = false;
+        let start = 0;
+        while (!isLastPage) {
+            const pageResult = await this.runQuery<BitbucketServer.Paginated<BitbucketServer.Repository>>(
+                userOrToken,
+                `/repos?${permission}start=${start}`,
+            );
+            if (pageResult.values) {
+                result.push(...pageResult.values);
             }
-            if (segments.length > 0) {
-                q = `?${segments.join("&")}`;
+            isLastPage = !!pageResult.isLastPage;
```

Comment:
TODO: `isLastPage` should default to `true` if `pageResult.isLastPage` is undefined.

Reply:
going to change to: `isLastPage = typeof === "undefined" || !!pageResult.isLastPage; // test for "undefined" is a fuse`